src/OSPSuite.SimModel/Simulation.cs

```diff
@@ -599,6 +599,11 @@ public void Dispose()
          GC.SuppressFinalize(this);
       }
 
+      ~Simulation()
+      {
+         Dispose(false);
+      }
```

Comment:
This is actually the only change. 
All other modifications are done to enable memory leak analysis on demand (and affect only the _Debug_ version of SimModelNative) . I will write a small howto later on here: 
https://github.com/Open-Systems-Pharmacology/OSPSuite.SimModel/wiki/Find-memory-leaks-with-Visual-Leak-Detector